Pattern found in the Cloud SDK version file could not be parsed: 218.0.0.213974472

com.google.cloud.tools.appengine.cloudsdk.CloudSdkVersionFileParseException: Pattern found in the Cloud SDK version file could not be parsed: 218.0.0.213974472
	at com.google.cloud.tools.appengine.cloudsdk.CloudSdk.getVersion(CloudSdk.java:95)
	at com.google.cloud.tools.appengine.cloudsdk.CloudSdk.validateCloudSdkVersion(CloudSdk.java:187)
	at com.google.cloud.tools.appengine.cloudsdk.CloudSdk.validateCloudSdk(CloudSdk.java:181)
	at com.google.cloud.tools.appengine.cloudsdk.GcloudRunner.run(GcloudRunner.java:75)
	at com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDeployment.deploy(CloudSdkAppEngineDeployment.java:88)
	at com.google.cloud.tools.intellij.appengine.java.cloud.AppEngineDeploy.deploy(AppEngineDeploy.java:152)
	at com.google.cloud.tools.intellij.appengine.java.cloud.executor.AppEngineStandardDeployTask.lambda$deploy$0(AppEngineStandardDeployTask.java:126)
	at com.google.cloud.tools.appengine.cloudsdk.process.LegacyProcessHandler.syncRun(LegacyProcessHandler.java:144)
	at com.google.cloud.tools.appengine.cloudsdk.process.LegacyProcessHandler.access$200(LegacyProcessHandler.java:31)
	at com.google.cloud.tools.appengine.cloudsdk.process.LegacyProcessHandler$3.run(LegacyProcessHandler.java:163)
Caused by: java.lang.IllegalArgumentException: Pattern "218.0.0.213974472" is not a valid CloudSdkVersion.
	at com.google.cloud.tools.appengine.cloudsdk.serialization.CloudSdkVersion.<init>(CloudSdkVersion.java:66)
	at com.google.cloud.tools.appengine.cloudsdk.CloudSdk.getVersion(CloudSdk.java:91)
	... 9 more
Exception class name:
java.lang.IllegalArgumentException
Exception message:
Pattern found in the Cloud SDK version file could not be parsed: 218.0.0.213974472
Throw file name:
CloudSdk.java
Throw line number:
95
Throw class name:
com.google.cloud.tools.appengine.cloudsdk.CloudSdk
Throw method name:
getVersion

Getting this error as well.

```
[ERROR] Failed to execute goal com.google.cloud.tools:appengine-maven-plugin:1.3.1:deploy (default-cli) on project helloworld: Execution default-cli of goal com.google.cloud.tools:appengine-maven-plugin:1.3.1:deploy failed: Pattern found in the Cloud SDK version file could not be parsed: 218.0.0.213974472: Pattern "218.0.0.213974472" is not a valid CloudSdkVersion. -> [Help 1]
```

Using the internal build of Cloud SDK. Can't deploy anything. Is there a fix?

--- a/src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java
+++ b/src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java
@@ -189,7 +189,7 @@ public class CloudSdk {
         throw new CloudSdkOutOfDateException(version, MINIMUM_VERSION);
       }
     } catch (CloudSdkVersionFileParseException ex) {
-      // unusual string in version file. Continue onward and hope.
+      logger.log(Level.WARNING, ex.getMessage(), ex);
     } catch (CloudSdkVersionFileNotFoundException ex) {
       // this is likely a version of the Cloud SDK prior to when VERSION files were introduced
       throw new CloudSdkOutOfDateException(MINIMUM_VERSION);

--- a/src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java
+++ b/src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java
@@ -188,6 +188,8 @@ public class CloudSdk {
       if (version.compareTo(MINIMUM_VERSION) < 0) {
         throw new CloudSdkOutOfDateException(version, MINIMUM_VERSION);
       }
+    } catch (CloudSdkVersionFileParseException ex) {
+      // unusual string in version file. Continue onward and hope.
     } catch (CloudSdkVersionFileNotFoundException ex) {
       // this is likely a version of the Cloud SDK prior to when VERSION files were introduced
       throw new CloudSdkOutOfDateException(MINIMUM_VERSION);

--- a/src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkTest.java
+++ b/src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkTest.java
@@ -28,7 +28,6 @@ import static org.mockito.Mockito.when;
 
 import com.google.cloud.tools.appengine.cloudsdk.serialization.CloudSdkVersion;
 import com.google.common.io.Files;
-
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
@@ -95,7 +94,7 @@ public class CloudSdkTest {
   public void testGetVersion_fileContentInvalid() throws IOException {
     String fileContents = "this is not a valid version string";
     writeVersionFile(fileContents);
-    try { 
+    try {
       sdk.getVersion();
       fail();
     } catch (CloudSdkVersionFileException ex) {
@@ -104,11 +103,11 @@ public class CloudSdkTest {
           ex.getMessage());
     }
   }
-  
+
   @Test
-  public void testValidateCloudSdk_versionFileContentInvalid() 
+  public void testValidateCloudSdk_versionFileContentInvalid()
       throws IOException, CloudSdkNotFoundException, CloudSdkOutOfDateException,
-      CloudSdkVersionFileException {
+          CloudSdkVersionFileException {
     String fileContents = "this is not a valid version string";
     writeVersionFile(fileContents);
     root.resolve("bin").toFile().mkdir();

--- a/src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkTest.java
+++ b/src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkTest.java
@@ -28,6 +28,7 @@ import static org.mockito.Mockito.when;
 
 import com.google.cloud.tools.appengine.cloudsdk.serialization.CloudSdkVersion;
 import com.google.common.io.Files;
+
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
@@ -94,7 +95,7 @@ public class CloudSdkTest {
   public void testGetVersion_fileContentInvalid() throws IOException {
     String fileContents = "this is not a valid version string";
     writeVersionFile(fileContents);
-    try {
+    try { 
       sdk.getVersion();
       fail();
     } catch (CloudSdkVersionFileException ex) {
@@ -102,6 +103,20 @@ public class CloudSdkTest {
           "Pattern found in the Cloud SDK version file could not be parsed: " + fileContents,
           ex.getMessage());
     }
+  }
+  
+  @Test
+  public void testValidateCloudSdk_versionFileContentInvalid() 
+      throws IOException, CloudSdkNotFoundException, CloudSdkOutOfDateException,
+      CloudSdkVersionFileException {
+    String fileContents = "this is not a valid version string";
+    writeVersionFile(fileContents);
+    root.resolve("bin").toFile().mkdir();
+    // fake SDK contents
+    root.resolve("bin/gcloud").toFile().createNewFile();
+    root.resolve("bin/gcloud.cmd").toFile().createNewFile(); // for Windows
+    root.resolve("bin/dev_appserver.py").toFile().createNewFile();
+    sdk.validateCloudSdk();
   }
 
   @Test

